### ISSUE TITLE
Plotting on a log/logit scale overwrites axis inverting

### ISSUE THREAD

**Issue by lkluft** [matplotlib/matplotlib#14620]
### Bug report

**Bug summary**

When plotting on a log axis that has been inverted, the inverting is overwritten.

This happens for ``"log"`` and ``"logit"`` scales but not for ``"linear"`` or ``"symlog"``, which both work as expected.

**Code for reproduction**

```python
import matplotlib.pyplot as plt


fig, ax = plt.subplots()
ax.set_yscale('log')  # or 'logit'
ax.yaxis.set_inverted(True)

print(ax.yaxis.get_inverted())
ax.plot(range(50), range(50))
print(ax.yaxis.get_inverted())
```

**Actual outcome**

```
True
False
# and the plot is **not** inverted
```

**Expected outcome**

```
True
True
# and an inverted plot
```

**Matplotlib version**
  * Operating system: Linux and MacOS
  * Matplotlib version: 3.1.0
  * Python version: 3.7.3

Python and matplotlib have been installed using conda.

**Comment by lkluft** [comment]
I am not sure if this is related to #14615 (or even a duplicate?)

**Comment by ivanov** [comment]
I think it is indeed a duplicate, @lkluft, at least on v3.1.x branch, with the fixes for #14615, I do indeed get the expected `True True` outcome for the code snippet you provided. Thanks for filing this and 
happy hacking! :bowtie:
